std/math.d

```diff
@@ -575,10 +575,11 @@ enum real SQRT1_2 =    SQRT2/2;                               /** $(SQRT)$(HALF)
  * Limitations:
  *     Does not work correctly for signed intergal types and value `Num`.min.
  */
-auto abs(Num)(Num x) @nogc pure nothrow
+auto abs(Num)(Num) @nogc pure nothrow
 if ((is(Unqual!Num == short) || is(Unqual!Num == byte)) ||
     (is(typeof(Num.init >= 0)) && is(typeof(-Num.init))))
 {
+    alias x = _param_0;
```

Comment:
It looks like `_param_0` is an implementation detail rather than part of the spec.  We would need to make this part of the spec before establishing a pattern that uses it I think.

Reply:
Many things that are unnamed in D get a name. That's a huge leak of abstraction, and shouldn't happen, but does due to the compiler having lacking abstractions. [I had a go at solving some of the problems it introduces](https://github.com/dlang/dmd/pull/10834) a bit ago but hell broke loose. But if anything, I believe we want to move away from exposing compiler-generated names, instead of relying on it.

If one want to access unnamed parameters, it should be via a traits / is expression.

Reply:
And regarding documentation, since an identifier cannot start with a digit, we can just use the index. It would need a fix to DDOC, that's all.